app/javascript/modCenter/moderationArticles.jsx

```diff
@@ -0,0 +1,46 @@
+import { h, Component } from 'preact';
+// import { request } from "../utilities/http"
+import SingleArticle from './singleArticle';
+
+export class ModerationArticles extends Component {
+  state = {
+    articles: [],
+  };
+
+  componentWillMount() {
+    const container = document.getElementById('mod-index-list');
```

Comment:
Does the dataset of articles on this element exist before this component is rendered? If so the `componentWillMount()` method can be removed and all the articles can be part of the initia state in the constructor.

Reply:
Ohh interesting, yes it exists before the component renders. Will update.

Reply:
That's good. So load it up in the initial state in the constructor and you'll avoid an unnecessary rerender.

Reply:
To be clear rerenders happen, but in this case, it would definitely change the DOM, so might as well do it all at once.